### PR TITLE
Limit the number of publisher suggestions to 15

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/26598

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Subscribe to several sources (in brave news)
2. Confirm the number of suggestions is 15 or less
3. If the number of sources is less than 15, subscribe to more sources until the number of suggestions is 15
4. Add one more source. The number of suggestions should still be 15.

### CODE REVIEW

File: components/brave_today/browser/suggestions_controller.cc [hunk 1]
```diff
@@ -253,21 +255,28 @@ SuggestionsController::GetSuggestedPublisherIdsWithHistory(
...
   }
 
-  std::sort(result.begin(), result.end(),
+  std::sort(all_suggestions.begin(), all_suggestions.end(),
             [scores](const std::string& a_id, const std::string& b_id) {
               return scores.at(a_id) > scores.at(b_id);
             });
 
-  return result;
+  auto suggestions_count = std::min(all_suggestions.size(), kMaxSuggestions);
```
Comment (fallaciousreasoning) [inline components/brave_today/browser/suggestions_controller.cc:272]: @sangwoo108 is there a better way to do this? I basically want to truncate the sources to `kMaxSuggestions`

Comment (sangwoo108) [reply components/brave_today/browser/suggestions_controller.cc:272]: Cool, I think your code is describing itself well, so good enough. If you would want to simpler one, I think you might want to use std::vector.resize().

```
if (all_suggestions.size() > kMaxSuggestions)
  all_suggestions.resize(kMaxSuggestions);
```

File: components/brave_today/browser/suggestions_controller.cc [hunk 2]
```diff
@@ -41,6 +41,8 @@ constexpr double kSimilarSubscribedMax = 0.4;
 constexpr double kSimilarVisitedMin = 0;
 constexpr double kSimilarVisitedMax = 0.3;
 
+constexpr uint64_t kMaxSuggestions = 15;
+
```
Comment (sangwoo108) [inline components/brave_today/browser/suggestions_controller.cc:45]: nit: If this variable is used only by the new code, how about moving this variable into the scope?

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

File: components/brave_today/browser/suggestions_controller.cc (hunk 1 above)
Comment (fallaciousreasoning) [reply components/brave_today/browser/suggestions_controller.cc:272]: Perfect, this is exactly what I wanted. Thanks @sangwoo108!

**Review by petemill — APPROVED** [review]

**Pull request merged**